README.rst

```diff
@@ -17,16 +18,89 @@ high-throughput imaging experiment. The measurements are stored across thousands
+
+  [filenames]
+  image   = image.csv      #or: Image.csv
+  object  = object.csv     #or: Object.csv
+
+Cytominer-Database is currently limited to the following measurement file kinds:
+Cells.csv, Cytoplasm.csv, Nuclei.csv, Image.csv, Object.csv.
+The [filenames] section in the configuration file saves the correct basename of existing measurement files
+(this may be important in the case of inconsistent capitalization).
+# TODO: Are there any other reasons for the [filenames] section?
+
+The [database_engine] section
+-----------------------------
+
+.. code-block::
+
+  [ingestion_engine]
+  engine = Parquet      #or: SQLite
+
+The [database_engine] section specifies the backend. Possible key-value pairs are:
+**engine** = *SQLite* or **engine** = *Parquet*.
+[Potential ToDo: Delete Section and introduce a command flag]
+
+The [schema] section
+--------------------
+
+.. code-block::
+
+ [schema]
+ reference_option = sample         #or: path/to/reference/folder relative to source_directory
+ ref_fraction     = 1              #or: any decimal value in [0, 1]
+ type_conversion  = int2float      #or: all2string
+
+The [schema] section specifies how to manage incompatibilities in the table schema
+of the files.
+In that case, a Parquet file is fixed to a schema with which it was first opened,
+i.e. by the first file which is written (the reference file). To append the data
+of all .csv files of that file-kind, it is important to assure the reference file
+satisfies certain incompatibility requirements, e.g. does not miss any columns
+and all existing files can be automatically converted to the reference schema.
+This section is used only if the files are ingested to Parquet format and was
+developed to handle the special cases in which tables that cannot be concatenated automatically.
+
+There are two options for the key **reference_option**:
+The first option is to create a designated folder containing one .csv reference file for every kind of file ("Cytoplasm.csv", "Nuclei.csv", ...) and save the folder path in the config file as **reference_option** = *path/to/reference/folder*, where the path is relative to the source_directory from the ingest command. These reference files' schema will determine the schema of the Parquet file into which all .csv files of its' kind will be ingested, hence the reference files must carefully selected to be complete in the number of columns and possess correct column names and types.
+
+Alternatively, the reference files can be found automatically from a sampled subset of all existing files.
+This is the case if **reference_option** = *sample* is set.
+A subset of all files is sampled uniformly at random and the first table with
+the maximum number of columns among all sampled .csv files is chosen as the reference table.
+If this case, an additional key **ref_fraction** can be set, which specifies the fraction of files
+sampled among all files. The default value is **ref_fraction** = *1* , for which
```

Comment:
For the release, some default value well below 1 would be better, if we can assume that almost all files do not have any missing columns and there are many files to ingest. The previous default value was set 1 because the number of testing files is small and hence even one or two corrupt .csv's could potentially dominate the outcome, if the sampling fraction is small. I double-checked all testing files and set the new default value to = 0.3 (tests still pass).